docs/docs/concepts/durable_execution.md

```diff
@@ -0,0 +1,50 @@
+
+LangGraph's built-in [persistence](./persistence.md) layer provides durable execution for workflows, ensuring that the state of each execution step is saved to a durable store. This capability guarantees that if a workflow is interrupted—whether by a system failure or for [human-in-the-loop](./human_in_the_loop.md) interactions - it can be resumed from its last recorded state.
+
+!!! tip
+
+    If you are using LangGraph with a checkpointer, you already have durable execution enabled. You can pause and resume workflows at any point, even after interruptions or failures.
+    To make the most of durable execution, ensure that your workflow is designed to be [deterministic](#determinism-and-consistent-replay) and [idempotent](#idempotency) and warp
+    any side effects or non-deterministic operations inside [tasks](./functional_api.md#task).
+
+## Requirements
+
+To leverage durable execution in LangGraph, you need to:
+
+1. Enable [persistence](./persistence.md) in your workflow by specifying a [checkpointer](./persistence.md#checkpointer-libraries) that will save workflow progress.
+2. Specify a [thread identifier](./persistence.md#threads) when executing a workflow. This will track the execution history for a particular instance of the workflow.
+3. Wrap any non-deterministic operations (e.g., random number generation) or operations with side effects (e.g., file writes, API calls) inside [tasks][langgraph.func.task] to ensure that when a workflow is resumed, these operations are not repeated for the particular run, and instead their results are retrieved from the persistence layer. For more information, see [Determinism and Consistent Replay](#determinism-and-consistent-replay).
+
+## Determinism and Consistent Replay
+
+When you resume a workflow run, the code does **NOT** resume from the **same line of code** where execution stopped; instead, it will identify an appropriate [starting point](#starting-points-for-resuming-workflows-advanced) from which to pick up where it left off. This means that the workflow will replay all steps from the [starting point](#starting-points-for-resuming-workflows-advanced) until it reaches the point where it was stopped.
+
+As a result, when you're writing a workflow for durable execution, you should wrap any non-deterministic operations (e.g., random number generation) and any operations with side effects (e.g., file writes, API calls) inside [tasks](./functional_api.md#task).
+
+- **Avoid Repeating Work**  Place operations that produce side effects (such as logging, file writes, or network calls) after an interrupt in Graph API nodes or encapsulate them within [tasks](./functional_api.md#task). This prevents their unintended repetition when the workflow is resumed. LangGraph will look up information stored in the persistence layer about any **tasks** that were previously executed for the specific run and will swap in the results of those tasks instead of re-executing them.
```

Comment:
> encapsulate them within tasks

Or nodes/runnables?

Reply:
Nodes yes, runnables no?

I can add a clarification on nodes. I chose to describe `tasks` only with the intention to avoid confusing people with two ways of doing the same thing. `tasks` are lighter weight since you don't need to declare a node (which implies declaring a schema). Imagine that having to accommodate a lot of different side-effects each with its own payload that's not immediately derivable from the shared graph state.

Reply:
I generally agree with your remarks about `tasks` and why they're preferable over nodes. But since this is a concept doc, I figured it's best to paint a complete picture, else someone might've read this and thought that functional API is a requirement for durability.

Also re: runnables, I'm fine with leaving those off. I was just being a bit pedantic, as any langchain runnable can be used as a node